src/AlgebraicGeometry/Surfaces/EnriquesAut.jl

```diff
@@ -0,0 +1,908 @@
+
+  gens_Dplus = reduce(vcat, gens_Dplus)
+  ECtx.Dplus = gens_Dplus
+
+
+  Dplus_perp, _ = orthogonal_submodule(DSY, Dplus)
+  Dplus_perp,_ = snf(Dplus_perp)
+
+  # needed for hash
+  ECtx.Dplus_perp = reduce(vcat,[change_base_ring(GF(2), 2*solve(basis_matrix(SY), matrix(QQ, 1, 26, lift(i)); side=:left)) for i in gens(Dplus_perp)])
+
+
+  @vprint :EnriquesAuto 2 "done\n"
+  ECtx.membership_test = membership_test_as_group
+  #upper_bound = mass(ECtx)*length(initial_automorphisms(ECtx))
+  return ECtx
+end
+  
+@doc raw"""
+    EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat) -> EnriquesBorcherdsCtx 
+  
+Return a context object for Borcherds' method for Enriques surfaces. 
+  
+Let ``\pi: X \to Y`` be the universal cover of an Enriques surface and ``\epsilon`` the Enriques involution. 
+# Input: 
+- `SY2` -- the invariant lattice of the Enriques involution in the numerical lattice `SX` of ``X``. 
+"""
+function EnriquesBorcherdsCtx(SY2::ZZLat, SX::ZZLat; ample=nothing)
+  @req is_sublattice(SY2, SX) "SY2 must be a sublattice of SX"
+  tmp = rescale(SY2, 1//2)
+  @req is_unimodular(tmp) && is_even(tmp) && signature_tuple(tmp) == (1, 0, 9) "SY2 must be isomorphic to E10(2)"
+  B = solve(basis_matrix(SX), basis_matrix(SY2); side=:left)
+  L26, SX1, iSX1 = embed_in_unimodular(lattice(rational_span(SX)), 1, 25, primitive=true,even=true)
+  SY1 = lattice(ambient_space(SX1),solve(basis_matrix(SX),basis_matrix(SY2);side=:left)*basis_matrix(SX1))
+  if ample==nothing 
+    ample = ample_class(SX1)
+    # orthogonal projection of ample to SY1
+    B = basis_matrix(SY1)
+    GB = gram_matrix(ambient_space(SY1))*transpose(B)
+    ample = solve(B*GB, 2*ample*GB; side=:left)
+  end
+  @assert only(ample*gram_matrix(SY2)*transpose(ample))>0
+  @show ample
+  weyl, u =  borcherds_method_preprocessing(L26, SY1; ample)
+  return EnriquesBorcherdsCtx(SY1, SX1, L26, weyl)
+end
+
+@doc raw"""
+    enriques_surface_automorphism_group(SY2::ZZLat, SX::ZZLat;ample=nothing)
+    
+Return generators for the automorphism group of an Enriques surface.
+  
+Let ``\pi: X \to Y`` be the universal cover of an Enriques surface and ``\epsilon`` the Enriques involution. Let ``S_Y`` be the numerical lattice of ``Y``. 
+This function computes the image of the natural map 
+```math
+\varphi_Y \colon \mathrm{Aut}_{s}(Y) \to O(S_Y \otimes \mathbb{F}_2)
+```
+where ``\mathrm{Aut}_{s}(Y) \leq \mathrm{Aut}(Y)`` denotes the subgroup of semi-symplectic automorphisms.
+Note that the kernel of ``\varphi_Y`` is finite.
+
+See [BS22](@cite) [BS22*1](@cite) [BRS23](@cite) for background and algorithms. 
+
+# Input: 
+- `SY2` -- the invariant lattice of the Enriques involution in the numerical lattice `SX` of ``X``. 
+- `ample` -- optionally an ample class as a ``1\times x 10`` matrix representing an ample class w.r.t. the basis of `SY2`; if not given, some arbitrary Weyl-chamber is picked.
+"""
+function enriques_surface_automorphism_group(SY2::ZZLat, SX::ZZLat; ample::ZZMatrix=nothing)
```

Comment:
```suggestion
function enriques_surface_automorphism_group(SY2::ZZLat, SX::ZZLat; ample::Union{ZZMatrix,Nothing}=nothing)
```
this won't work otherwise